presto-main/src/main/java/io/prestosql/sql/planner/sanity/PartitionFilterChecker.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.sanity;
+
+import com.google.common.base.VerifyException;
+import io.prestosql.Session;
+import io.prestosql.execution.warnings.WarningCollector;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.metadata.TableHandle;
+import io.prestosql.metadata.TableProperties;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.sql.planner.TypeAnalyzer;
+import io.prestosql.sql.planner.TypeProvider;
+import io.prestosql.sql.planner.plan.PlanNode;
+import io.prestosql.sql.planner.plan.PlanVisitor;
+import io.prestosql.sql.planner.plan.TableScanNode;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.prestosql.SystemSessionProperties.isQueryPartitionFilterRequired;
+
+public class PartitionFilterChecker
```

Comment:
This check should be specific to the Hive connector and be implemented within the connector. There's no concept of table partitioning in the engine.

Reply:
Any suggestions on where this should be called int he Hive connector?